### PR TITLE
[Components with more than 1 service] LPS-188721 Refactor ModelListener components part II

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-188721

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:search

**Comment by liuxingye** [comment]
ci:test:stable

**Comment by liuxingye** [comment]
Hi @MrJohn1911,
This is a failure about [com.liferay.search.experiences.internal.model.listener.test.CompanyModelListenerTest](https://testray.liferay.com/home/-/testray/case_results/2366873961), could you take a look when you get a chance? Thanks.
```
     [exec] com.liferay.search.experiences.internal.model.listener.test.CompanyModelListenerTest > testSXPElements FAILED
     [exec]     java.lang.NoSuchFieldException: No field on null with name _sxpElements
     [exec]     	at com.liferay.portal.kernel.test.ReflectionTestUtil.getField(ReflectionTestUtil.java:135)
     [exec]     	at com.liferay.portal.kernel.test.ReflectionTestUtil.getFieldValue(ReflectionTestUtil.java:151)
     [exec]     	at com.liferay.search.experiences.internal.model.listener.test.CompanyModelListenerTest.testSXPElements(CompanyModelListenerTest.java:46)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:353)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$1.evaluate(TestExecutorRunnable.java:237)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
     [exec]     	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._execute(TestExecutorRunnable.java:316)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.run(TestExecutorRunnable.java:95)
     [exec]     	at java.lang.Thread.run(Thread.java:745)
```
Whole [log](https://testray.liferay.com/reports/production/logs/2023-06/test-1-37/test-portal-acceptance-pullrequest(master)/12803/modules-integration-mysql57-jdk8/0/11/jenkins-console.txt.gz?authuser=0).

cc @zhangyan0701

**Comment by liuxingye** [comment]
@slnn Could you help to add QA Failed Label based on the above ^ backend failure? Thanks.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:search

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by chaowang-123** [comment]
Hi @MrJohn1911 
There has an SF issue, could you fix it when you have the chance?
```
format-source-files:
     [java] java.lang.Exception: Found 4 formatting issues:
     [java] 1: Private non-static field '_SCHEMA_VERSION' must match pattern '^_[a-z0-9][_a-zA-Z0-9]*$', see https://checkstyle.sourceforge.io/config_naming.html#MemberName: ./modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/model/listener/CompanyModelListenerHelper.java 132 (Checkstyle:MemberNameCheck)
     [java] 2: ./modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/model/listener/CompanyModelListenerHelper.java expected:<...lves
     [java]  */
     [java] @Component([enabled = false, service = CompanyModelListenerHelper.class)
     [java] public class CompanyModelListenerHelper {
     [java] 
     [java] 	public void addSXPElements(
     [java] 			Company company, SXPElementLocalService sxpElementLocalService)
     [java] 		throws PortalException {
     [java] 
     [java] 		Set<String> externalReferenceCodes = new HashSet<>();
     [java] 
     [java] 		for (com.liferay.search.experiences.model.SXPElement sxpPElement :
     [java] 				sxpElementLocalService.getSXPElements(
     [java] 					company.getCompanyId(), true)) {
     [java] 
     [java] 			externalReferenceCodes.add(sxpPElement.getExternalReferenceCode());
     [java] 		}
     [java] 
     [java] 		for (SXPElement sxpElement : _getSXPElements()) {
     [java] 			if ((!FeatureFlagManagerUtil.isEnabled("LPS-122920") &&
     [java] 				 Objects.equals(
     [java] 					 sxpElement.getExternalReferenceCode(),
     [java] 					 "RESCORE_BY_TEXT_EMBEDDING")) ||
     [java] 				externalReferenceCodes.contains(
     [java] 					sxpElement.getExternalReferenceCode())) {
     [java] 
     [java] 				continue;
     [java] 			}
     [java] 
     [java] 			User user = company.getGuestUser();
     [java] 
     [java] 			sxpElementLocalService.addSXPElement(
     [java] 				sxpElement.getExternalReferenceCode(), user.getUserId(),
     [java] 				LocalizedMapUtil.getLocalizedMap(
     [java] 					sxpElement.getDescription_i18n()),
     [java] 				String.valueOf(sxpElement.getElementDefinition()), true,
     [java] 				_SCHEMA_VERSION,
     [java] 				LocalizedMapUtil.getLocalizedMap(sxpElement.getTitle_i18n()), 0,
     [java] 				new ServiceContext() {
     [java] 					{
     [java] 						setAddGuestPermissions(true);
     [java] 						setCompanyId(company.getCompanyId());
     [java] 						setScopeGroupId(company.getGroupId());
     [java] 						setUserId(user.getUserId());
     [java] 					}
     [java] 				});
     [java] 		}
     [java] 	}
```

whole [log](https://testray.liferay.com/reports/production/logs/2023-07/test-1-39/test-portal-source-format/3332/source-format-current-branch-jdk8/0/0/jenkins-console.txt.gz?authuser=0).

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
no regressions found

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/model/listener/CompanyModelListenerHelper.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(enabled = false, service = CompanyModelListenerHelper.class)
+public class CompanyModelListenerHelper {
```
Comment (dantewang) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/model/listener/CompanyModelListenerHelper.java:49]: @MrJohn1911 I think this class is more focused on `SXPElement`, so maybe rename it to `SXPElementHelper` and put it in another package. This also makes it less weird when you apply the component to the `SXPServiceVerifyProcess`.

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/model/listener/test/CompanyModelListenerTest.java [hunk 2]
```diff
@@ -43,8 +43,11 @@ public class CompanyModelListenerTest {
 
 	@Test
 	public void testSXPElements() throws Exception {
-		List<SXPElement> sxpElements = ReflectionTestUtil.getFieldValue(
-			_modelListener, "_sxpElements");
+		Object companyModelListenerHelper = ReflectionTestUtil.getFieldValue(
```
Comment (dantewang) [inline modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/model/listener/test/CompanyModelListenerTest.java:46]: @MrJohn1911 This test only tests the `SXPElements` related thing in `CompanyModelListener`. When you extract these logic out, the test should be also altered to test that new class directly.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@MrJohn1911 Please see my comment and resend, thanks!

**Pull request closed without merging**